src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/controllers/v1/person/AlertsControllerTest.kt

```diff
@@ -0,0 +1,162 @@
+
+      it("retrieves the alerts for a person with the matching ID") {
+        mockMvc.perform(MockMvcRequestBuilders.get("$path")).andReturn()
+
+        verify(getAlertsForPersonService, VerificationModeFactory.times(1)).execute(pncId)
+      }
+
+      it("returns the alerts for a person with the matching ID") {
+        val result = mockMvc.perform(MockMvcRequestBuilders.get("$path")).andReturn()
+
+        result.response.contentAsString.shouldContain(
+          """
+          "data": [
+            {
+              "offenderNo": "A7777ZZ",
+              "type": "X",
+              "typeDescription": "Security",
+              "code": "XNR",
+              "codeDescription": "Not For Release",
+              "comment": "IS91",
+              "createDate": "2022-08-01",
+              "expiryDate": "2023-08-01",
+              "expired": false,
+              "active": true
+            }
+          ]
+        """.removeWhitespaceAndNewlines(),
+        )
+      }
+
+      it("returns an empty list embedded in a JSON object when no alerts are found") {
+        val pncIdForPersonWithNoAlerts = "0000/11111A"
+        val encodedPncIdForPersonWithNoAlerts =
+          URLEncoder.encode(pncIdForPersonWithNoAlerts, StandardCharsets.UTF_8)
+        val path = "/v1/persons/$encodedPncIdForPersonWithNoAlerts/alerts"
+
+        whenever(getAlertsForPersonService.execute(pncIdForPersonWithNoAlerts)).thenReturn(
+          Response(
+            data = emptyList(),
+          ),
+        )
+
+        val result =
+          mockMvc.perform(MockMvcRequestBuilders.get("$path"))
+            .andReturn()
+
+        result.response.contentAsString.shouldContain("\"data\":[]".removeWhitespaceAndNewlines())
+      }
+
+      it("responds with a 404 NOT FOUND status when person isn't found in the upstream API") {
+        whenever(getAlertsForPersonService.execute(pncId)).thenReturn(
+          Response(
+            data = emptyList(),
+            errors = listOf(
+              UpstreamApiError(
+                causedBy = UpstreamApi.NOMIS,
+                type = UpstreamApiError.Type.ENTITY_NOT_FOUND,
+              ),
+            ),
+          ),
+        )
+
+        val result = mockMvc.perform(MockMvcRequestBuilders.get("$path")).andReturn()
+
+        result.response.status.shouldBe(HttpStatus.NOT_FOUND.value())
+      }
+
+      it("returns paginated results") {
+        whenever(getAlertsForPersonService.execute(pncId)).thenReturn(
+          Response(
+            data =
+            List(20) {
+              Alert(
+                offenderNo = "A7777ZZ",
+                type = "X",
+                typeDescription = "Security",
+                code = "XNR",
+                codeDescription = "Not For Release",
+                comment = "IS91",
+                createDate = LocalDate.parse("2022-08-01"),
+                expiryDate = LocalDate.parse("2023-08-01"),
+                expired = false,
+                active = true,
```

Comment:
I am happy this has captured the full scope of the data fields, and that 'Alert' has been removed from 'type', 'typeDescription', 'code' and 'codeDescription'